corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -553,6 +553,40 @@ hqDefine("cloudcare/js/formplayer/app", function () {
         $.ajax(options);
     });
 
+    FormplayerFrontend.getChannel().reply("interval_sync-db", function (unusedOptions) {
+        var user = FormplayerFrontend.getChannel().request('currentUser'),
+            username = user.username,
+            domain = user.domain,
+            formplayerUrl = user.formplayer_url,
+            complete,
+            data = {
+                "username": username,
+                "domain": domain,
+                "restoreAs": user.restoreAs,
+                "app_id": unusedOptions.appId,
+            },
+            options;
+
+        complete = function (response) {
+            if (response.status === 'retry') {
+                FormplayerFrontend.trigger('retry', response, function () {
+                    options.data = JSON.stringify($.extend(true, {mustRestore: true}, data));
+                    $.ajax(options);
+                }, gettext('Waiting for server progress'));
+            } else {
+                FormplayerFrontend.trigger('clearProgress');
+                CloudcareUtils.formplayerSyncComplete(response.responseJSON.status === 'error');
```

Comment:
I'm not sure I understand this line. Doesn't this mean any completed sync will always show as an error?